apps/test/unit/templates/TooltipOverlayTest.js

```diff
@@ -17,148 +17,113 @@ describe('TooltipOverlay', () => {
-                />
-                <text x={EXPECTED_TEXT_X} y={187} style={styles.text}>two</text>
-              </g>
-            </g>
-        );
-    expect(EXPECTED_RECT_X).to.be.below(TEST_APP_WIDTH);
-    expect(EXPECTED_TEXT_X).to.be.below(TEST_APP_WIDTH);
-    expect(TEST_APP_HEIGHT).to.be.above(187);
+    const tooltip = shallowRender(TEST_APP_WIDTH, TEST_APP_HEIGHT, [textProvider('one'), textProvider('two')]);
+    const EXPECTED_X = 185;
+    validateRect(tooltip.find('rect').first(), EXPECTED_X, 148);
+    validateRect(tooltip.find('rect').last(), EXPECTED_X, 173);
+    validateText(tooltip.find('text').first(), EXPECTED_X, 148, 'one');
+    validateText(tooltip.find('text').last(), EXPECTED_X, 173, 'two');
   });
 
   it('generates a set of tooltip strings from providers that return a string', () => {
-    expect(withProviders([]).getTooltipStrings()).to.be.empty;
+    expect(withProviders([]).instance().getTooltipStrings()).to.be.empty;
 
     expect(withProviders([
       () => '',
       () => null,
       () => null
-    ]).getTooltipStrings().length).to.equal(1);
+    ]).instance().getTooltipStrings().length).to.equal(1);
 
     expect(withProviders([
       () => 'one',
       () => 'two',
       () => 'three'
-    ]).getTooltipStrings().length).to.equal(3);
+    ]).instance().getTooltipStrings().length).to.equal(3);
   });
 
   it('generates tooltip dimensions that depend on the length of the string set', () => {
-    expect(withStrings([]).getTooltipDimensions()).to.deep.equal({
+    expect(withStrings([]).instance().getTooltipDimensions()).to.deep.equal({
       width: TEXT_RECT_WIDTH,
       height: 0
     });
 
-    expect(withStrings(['one']).getTooltipDimensions()).to.deep.equal({
+    expect(withStrings(['one']).instance().getTooltipDimensions()).to.deep.equal({
       width: TEXT_RECT_WIDTH,
       height: TEXT_RECT_HEIGHT
     });
 
-    expect(withStrings(['one', 'two']).getTooltipDimensions()).to.deep.equal({
+    expect(withStrings(['one', 'two']).instance().getTooltipDimensions()).to.deep.equal({
       width: TEXT_RECT_WIDTH,
       height: 2 * TEXT_RECT_HEIGHT + BETWEEN_RECT_MARGIN
     });
 
-    expect(withStrings(['one', 'two', 'three']).getTooltipDimensions()).to.deep.equal({
+    expect(withStrings(['one', 'two', 'three']).instance().getTooltipDimensions()).to.deep.equal({
       width: TEXT_RECT_WIDTH,
       height: 3 * TEXT_RECT_HEIGHT + 2 * BETWEEN_RECT_MARGIN
     });
   });
 
+  describe('coordinatesProvider', function () {
+    it('maps props mouseX and mouseY to a coordinate string', function () {
+      const props = {
+        mouseX: 50,
+        mouseY: 100
+      };
+      expect(coordinatesProvider()(props)).to.equal(`x: ${props.mouseX}, y: ${props.mouseY}`);
+    });
+
+    it('floors the given coordinates', function () {
+      const props = {
+        mouseX: 1.3,
+        mouseY: 2.9
+      };
+      expect(coordinatesProvider()(props)).to.equal('x: 1, y: 2');
+    });
+  });
```

Comment:
in the future, it can be helpful to move code in a separate commit, so that it's easier to see what changes there are in the code that moved by looking at the other commits. not a big deal, just a best practice of sorts. :-)

Reply:
Thanks! I'll do that in the future